cmd/kubeadm/app/cmd/version.go

```diff
@@ -35,10 +44,39 @@ func NewCmdVersion(out io.Writer) *cobra.Command {
 			kubeadmutil.CheckErr(err)
 		},
 	}
+	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().StringP("output", "o", "", "output format, options available are yaml and json")
 	return cmd
 }
 
 func RunVersion(out io.Writer, cmd *cobra.Command) error {
-	fmt.Fprintf(out, "kubeadm version: %#v\n", version.Get())
+	v := Version{}
+
+	clientVersion := version.Get()
+	v.ClientVersion = &clientVersion
+
+	switch of := cmdutil.GetFlagString(cmd, "output"); of {
+	case "":
+		if cmdutil.GetFlagBool(cmd, "short") {
+			fmt.Fprintf(out, "Kubeadm Version: %s\n", clientVersion.GitVersion)
+		} else {
+			fmt.Fprintf(out, "Kubeadm Version: %s\n", fmt.Sprintf("%#v", clientVersion))
```

Comment:
nit: I prefer `kubeadm version: foo`. We don't write Kubeadm anywhere else, I've only seen it lowercased